packages/framework/src/Model/Order/Processing/OrderProcessorMiddleware/ApplyNominalPromoCodeMiddleware.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessorMiddleware;
+
+use Shopsys\FrameworkBundle\Component\Domain\Config\DomainConfig;
+use Shopsys\FrameworkBundle\Component\Money\Money;
+use Shopsys\FrameworkBundle\Component\Translation\Translator;
+use Shopsys\FrameworkBundle\Model\Order\Item\OrderItemData;
+use Shopsys\FrameworkBundle\Model\Order\Item\OrderItemDataFactory;
+use Shopsys\FrameworkBundle\Model\Order\Item\OrderItemTypeEnum;
+use Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessingData;
+use Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessingStack;
+use Shopsys\FrameworkBundle\Model\Order\PromoCode\DiscountCalculation;
+use Shopsys\FrameworkBundle\Model\Order\PromoCode\Exception\PromoCodeException;
+use Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCode;
+use Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCodeFacade;
+use Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCodeLimit\PromoCodeLimit;
+use Shopsys\FrameworkBundle\Model\Pricing\Vat\VatFacade;
+use Shopsys\FrameworkBundle\Twig\PriceExtension;
+
+class ApplyNominalPromoCodeMiddleware implements OrderProcessorMiddlewareInterface
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Item\OrderItemDataFactory $orderItemDataFactory
+     * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\PromoCodeFacade $promoCodeFacade
+     * @param \Shopsys\FrameworkBundle\Model\Order\PromoCode\DiscountCalculation $discountCalculation
+     * @param \Shopsys\FrameworkBundle\Model\Pricing\Vat\VatFacade $vatFacade
+     * @param \Shopsys\FrameworkBundle\Twig\PriceExtension $priceExtension
+     */
+    public function __construct(
+        protected readonly OrderItemDataFactory $orderItemDataFactory,
+        protected readonly PromoCodeFacade $promoCodeFacade,
+        protected readonly DiscountCalculation $discountCalculation,
+        protected readonly VatFacade $vatFacade,
+        protected readonly PriceExtension $priceExtension,
+    ) {
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessingData $orderProcessingData
+     * @param \Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessingStack $orderProcessingStack
+     * @return \Shopsys\FrameworkBundle\Model\Order\Processing\OrderProcessingData
+     */
+    public function handle(
+        OrderProcessingData $orderProcessingData,
+        OrderProcessingStack $orderProcessingStack,
+    ): OrderProcessingData {
+        $appliedPromoCodes = $orderProcessingData->orderInput->getPromoCodes();
+
+        $orderData = $orderProcessingData->orderData;
+
+        foreach ($appliedPromoCodes as $appliedPromoCode) {
+            if ($appliedPromoCode->getDiscountType() !== PromoCode::DISCOUNT_TYPE_NOMINAL) {
+                continue;
+            }
+
```

Comment:
do not we need to call validatePromoCode here?

Reply:
Yes, yes we need... Thanks :) aee85d3cd0